rfc/0014-metadata.md

```diff
@@ -0,0 +1,616 @@
+
+### keywords
+
+This attribute is considered optional by Artifact Hub:
+
+```yaml
+keywords:
+- pod
+- signature
+- sigstore
+- trusted
+```
+
+This value will be taken from the following annotation of `metadata.yml`:
+
+```yaml
+annotations:
+  io.artifacthub.keywords: "pod, signature, sigstore, trusted"
+```
+
+Given `annotations` is a map with both keys and values being strings, we have to join
+the keywords using the `,` symbol.
+
+Given all the annotations inside of `metadata.yml` are optional, the `metadata.yml`
+file might be missing this value. This value is not required by `artifacthub-pkg.yml`,
+hence `kwctl scaffold` not fail if this annotation is not found inside of `metadata.yml`.
+
+### `links`
+
+This attribute is considered optional by Artifact Hub:
+
+```yaml
+links:
+- name: policy
+  url: https://github.com/kubewarden/verify-image-signatures/releases/download/v0.1.6/policy.wasm
+- name: source
+  url: https://github.com/kubewarden/verify-image-signatures
+```
+
+The link pointing to the `wasm` module file is going to be computed when the 
+value of the `io.kubewarden.policy.source` indicates that the policy is hosted on GitHub.
+We will assume the `.wasm` policy is released at the following url:
+
+```rust
+let url = format!("{}/releases/download/v{}/policy.wasm", sourceUrl, version);
+```
+Instead, the value of the `source` link is going to be built using the
+following annotation:
+
+```yaml
+annotations:
+  io.kubewarden.policy.source: https://github.com/kubewarden/verify-image-signatures
+```
+
+Given all the annotations inside of `metadata.yml` are optional, the `metadata.yml`
+file might be missing this value. This value is not required by `artifacthub-pkg.yml`,
+hence `kwctl scaffold` not fail if this annotation is not found inside of `metadata.yml`.
+
+### Hard coded values
+
+There are some values that are going to be always the same, regardless of the policy and
+its version:
+
+```yaml
+provider:
+  name: kubewarden
+recommendations:
+- url: https://artifacthub.io/packages/helm/kubewarden/kubewarden-controller
+```
+
+These values are going to be hard coded inside of kwctl.
+
+### Kubewarden specific annotations``
+
+The `artifacthub-pkg.yml` has its own `annotations` field. This is optional and
+is a dictionary with strings both keys and values.
+
+We are always going to have this data inside of it, since it comes from
+values that are mandatory inside of `metadata.yml`:
+
+```yaml
+annotations:
+  kubewarden/mutation: true
+  kubewarden/contextAware: false
+```
+
+These value will be taken from the following keys of `metadata.yml`:
+
+```yaml
+mutating: true
+contextAware: false
+```
+
+We are also going to attempt to create the following annotation inside of `artifacthub-pkg.yml`:
+
+```yaml
+annotations:
+  kubewarden/resources: Pod, Deployment
+```
+This is going to be created by copying the value of this `metadata.yml` annotation:
+
+```yaml
+annotations:
+  io.artifacthub.resources: Pod, Deployment
```

Comment:
this is a new `metadata.yml` annotation, duplicating `rules.resources` yet in uppercase, singular, etc, correct?

Reply:
yes